modules/postgis/src/main/java/org/testcontainers/containers/PostgisContainer.java

```diff
@@ -0,0 +1,103 @@
+package org.testcontainers.containers;
+
+import org.jetbrains.annotations.NotNull;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.Set;
+
+import static java.time.temporal.ChronoUnit.SECONDS;
+
+/**
+ * @author wimdeblauwe
+ */
+public class PostgisContainer<SELF extends PostgisContainer<SELF>> extends JdbcDatabaseContainer<SELF> {
```

Comment:
I wonder if it's possible to re-implement it on top of `PostgresContainer` with something like `withGIS(true)`?

Reply:
That seems possible I think. I can create a new branch to give it a try. How would you like it to be done?

I assume returning `true` for `postgis` as well in `PostgreSQLContainerProvider` is the first step:

```
@Override
    public boolean supports(String databaseType) {
        return databaseType.equals(PostgreSQLContainer.NAME);
    }
```
Do I create a new constant `PostgreSQLContainer.NAME_POSTGIS` or `PostgreSQLContainer.POSTGIS_NAME`? Do I rename the `NAME` to `NAME_POSTGRES` or `POSTGRES_NAME` or leave it like it is?

As for `PostgreSQLContainer` itself. I could add a `boolean` flag to the constructor and have 1 class. Or I could create 2 subclasses and an abstract base class. Do you have any preference?

If we go with 2 subclasses, then the naming question above is maybe off the table, since then it could be `PostgisSQLContainer.NAME`?

Reply:
@bsideup Can you comment on how you want to proceed with this PR ?

Reply:
Hi @wimdeblauwe,

Please excuse me for not replying for a long time, some full time employment's crazy releases going on :D 

Maybe @rnorth or @kiview could help here. I'll do my best to find time to review as well 👍

Reply:
Hey @wimdeblauwe, sorry for leaving you hanging in the air for some time, I'll try to finish this PR up with you now.

I think it's a good idea to simply extend the functionality of `PostgreSQLContainer` it's fine to add an additional builder like setter (i.e. `withGis(true)`, like @bsideup proposed). 

Regarding `PostgreSQLContainerProvider`, I think it's much easier and less invasive, to add an additional `PostgisContainerProvider` class, WDYT?

Reply:
Ok for me. I created https://github.com/testcontainers/testcontainers-java/pull/708 as an alternative. If that one is ok, you can decline this PR and accept the other. Or I can put the changes from the other on this branch if you like that more.